test/functional/terminal/buffer_spec.lua

```diff
@@ -255,8 +261,14 @@ describe(':terminal buffer', function()
     command('bdelete!')
   end)
 
-  it('handles wqall', function()
+  it("requires bang (!) to close a running job", function()
+    local cwd = funcs.fnamemodify('.', ':p:~'):gsub([[[\/]*$]], '')
+    local ext_pat = iswin() and '%.EXE' or ''
     eq('Vim(wqall):E948: Job still running', exc_exec('wqall'))
+    matches('^Vim%(bdelete%):E89: term://'..pesc(cwd)..'//%d+:'..nvim_dir..'/tty%-test'..ext_pat..' will be killed %(add %! to override%)$', exc_exec('bdelete'))
+    command('call jobstop(&channel)')
+    retry(nil, nil, function() neq(-1, eval('jobwait([&channel], 0)[0]')) end)
```

Comment:
-1 doesn't seem like the right thing to check. `:help jobwait`

> -1 if the timeout was exceeded
> -2 if the job was interrupted (by |CTRL-C|)
> -3 if the job-id is invalid

thus we should check eq(-3, ...)
```suggestion
    retry(nil, nil, function() eq(-3, eval('jobwait([&channel], 0)[0]')) end)
```